File: src/tests/end_to_end/idle/idle_test.cc
// Copyright 2023 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <iostream>

#include <gtest/gtest.h>

namespace {

TEST(IdleTest, Idle4Mins) {
  for (int i = 0; i < 4; i++) {
    std::cout << "1 minute sleep (" << i + 1 << "/4)..." << std::endl;
    sleep(60);
  }
  std::cout << "Wake up!" << std::endl;
}

}  // namespace

int main(int argc, char** argv) {
  testing::InitGoogleTest(&argc, argv);
  return RUN_ALL_TESTS();
}

File: src/tests/end_to_end/idle/idle_test.go
// Copyright 2023 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package idletest

import (
	"fmt"
	"testing"
	"time"
)

func TestIdle4Mins(t *testing.T) {
	mpower, err := NewPowerMeasurement("Idle4Mins")
	if err != nil {
		t.Fatalf("NewPowerMeasurement: %s", err)
	}
	defer func() {
		if err := mpower.Stop(); err != nil {
			t.Fatalf("mpower.Stop: %s", err)
		}
	}()
	for i := 0; i < 4; i++ {
		fmt.Printf("1 minute sleep (%d/4)...\n", i+1)
		time.Sleep(time.Minute)
	}
	fmt.Println("Wake up!")
}
